### ISSUE TITLE
Using partial in combination with a defaultdict fails

### ISSUE THREAD

**Issue by lovetox** [microsoft/pyright#3834]
**Describe the bug**

using partial in combination with a defaultdict fails with

```
Argument type is partially unknown 
Argument corresponds to parameter "__default_factory" in function "
Argument type is "partial[deque[Unknown]]"
```

**To Reproduce**
```python
from collections import defaultdict
from collections import deque
from functools import partial

mydict: dict[str, deque[str]] = defaultdict(partial(deque, maxlen=100))
```

**Additional context**
Version 1.1.266

**Comment by erictraut** [comment]
The `partial` constructor call returns a value of type `partial[deque[Unknown]]`. You apparently have the strict-mode `reportUnknownArgumentType` diagnostic check enabled, so this is flagged as a partially-unknown type.

So this is working correctly. You'll need to find some workaround if you want to use this pattern along with `reportUnknownArgumentType`.
